subsys/bluetooth/audio/cap_initiator.c

```diff
@@ -1692,34 +1748,259 @@ int bt_cap_initiator_unicast_audio_stop(const struct bt_cap_unicast_audio_stop_p
 
 	for (size_t i = 0U; i < param->count; i++) {
 		struct bt_cap_stream *cap_stream = param->streams[i];
+		struct bt_bap_stream *bap_stream = &cap_stream->bap_stream;
+
+		/* Ensure that ops are registered before any procedures are started */
+		bt_cap_stream_ops_register_bap(cap_stream);
 
 		active_proc->proc_param.initiator[i].stream = cap_stream;
-	}
+		active_proc->proc_param.initiator[i].stop.release = param->release;
 
-	bt_cap_common_start_proc(BT_CAP_COMMON_PROC_TYPE_STOP, param->count);
+		if (can_disable_stream(bap_stream)) {
+			can_disable = true;
+		}
```

Comment:
Looking at the code again it's because it's in a loop. 

If any stream can be disabled/stopped/released we set the variable. Doing `can_disable = can_disable_stream(bap_stream);` would set the value based on the last stream we iterate on.